src/components/hack-details.js

```diff
@@ -13,88 +23,161 @@ const useStyles = makeStyles(theme => ({
+  const joinHack = event => {
+    const object = {
+      hackId: props.match.params.id,
+      userId: props.userId
+    };
+    Axios.post(process.env.REACT_APP_API_URL + "joinhack", object).then(res => {
+      setTeam(res.data.team);
+    });
+  };
+
+  const joinHackButton = () => {
+    if (team !== undefined) {
+      if (team.includes(props.userId)) {
+        return (
+          <Button variant="contained" color="primary" onClick={joinHack} disabled>
+            Join Hack
+          </Button>
+        );
+      } else {
+        return (
+          <Button variant="contained" color="primary" onClick={joinHack}>
+            Join Hack
+          </Button>
+        );
+      }
+    }
+    return(
+      <Button variant="contained" color="primary" onClick={joinHack}>
+            Join Hack
+          </Button>
+    )
+    
+  };
+
+  const showTeam = () => {
+    if (team !== undefined) {
+      return (
+        <Grid item xs={12}>
+          <Typography variant="h6">Team:</Typography>
           <Grid item xs={10}>
-            <Typography variant="body1">{goal}</Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant="h6">Team:</Typography>
+            {team.map(id => {
+              return (
+                <Chip
+                  key={id}
+                  className={classes.chipMArgin}
+                  icon={<AccountCircleIcon />}
+                  label={id}
+                  color="primary"
+                />
+              );
+            })}
           </Grid>
         </Grid>
-      </Paper>
-    </Grid>
-  );
+      );
+    }
+  };
+
+  if (title !== "") {
+    return (
+      <Grid
+        container
+        direction="column"
+        justify="center"
+        alignItems="stretch"
+        alignContent="center"
+      >
+        <Paper className={classes.root}>
+          <Grid
+            container
+            justify="space-between"
+            alignItems="center"
+            alignContent="center"
+            spacing={1}
+          >
+            <Grid item xs={9}>
+              <Chip
+                icon={<ErrorOutlineOutlinedIcon />}
+                label="status"
+                color="secondary"
+              />
+            </Grid>
+            <Grid item xs={3} className={classes.rightFeild}>
+              {joinHackButton()}
+            </Grid>
+            <Grid item xs={12}>
+              <Typography variant="h4">{title}</Typography>
+            </Grid>
+            <Grid item xs={12}>
+              <Typography variant="h6">Idea:</Typography>
+            </Grid>
+            <Grid item xs={10}>
+              <Typography variant="body1">{description}</Typography>
+            </Grid>
+            <Grid item xs={12}>
+              <Typography variant="h6">Goal:</Typography>
+            </Grid>
+            <Grid item xs={10}>
+              <Typography variant="body1">{goal}</Typography>
+            </Grid>
+            {showTeam()}
```

Comment:
there is no reason to have this this way, and may (although probably not in this case) cause performance problems by not maintain referential integrity of your components (what react uses to make performance optimisations), you'd be better off defining a new component and passing down the props it needs

```
<Team team={team} goal={goal} />
```

This will allow react to optimise better when and how much of the component tree it renders based on props changing, and then how much work it needs to do with the real dom.

Reply:
Team's has been moved to a component